server/pulp/server/db/connection.py

```diff
@@ -186,18 +216,9 @@ class PulpCollection(Collection):
     applications
     """
 
-    _decorated_methods = ('get_lasterror_options', 'set_lasterror_options',
-                          'unset_lasterror_options', 'insert', 'save', 'update', 'remove', 'drop',
-                          'find', 'find_one', 'count', 'create_index', 'ensure_index',
-                          'drop_index', 'drop_indexes', 'reindex', 'index_information', 'options',
-                          'group', 'rename', 'distinct', 'map_reduce', 'inline_map_reduce',
-                          'find_and_modify')
-
     def __init__(self, database, name, create=False, **kwargs):
         super(PulpCollection, self).__init__(database, name, create=create, **kwargs)
-
-        for m in self._decorated_methods:
-            setattr(self, m, retry_decorator(self.full_name)(getattr(self, m)))
+        UnsafeRetry().decorate_instance(instance=self, full_name=self.full_name)
```

Comment:
UnsafeRetry can become an object with only @classmethods on it since self is never meaningfully referenced.